solarforecastarbiter/metrics/tests/test_calculator.py

```diff
@@ -312,6 +373,325 @@ def test_calculate_deterministic_metrics(categories, metrics,
+                                        pd.DataFrame(np.random.randn(10, 3)),
+                                        pd.Series(np.random.randn(10)))
+    with pytest.raises(RuntimeError):
+        calculator.calculate_probabilistic_metrics(
+            proc_fxobs, LIST_OF_CATEGORIES, probabilistic._REQ_REF_FX
+        )
+
+
+def test_calculate_probabilistic_metrics_no_reference(
+        single_prob_forecast_observation, create_processed_fxobs):
+    proc_fxobs = create_processed_fxobs(single_prob_forecast_observation,
+                                        pd.DataFrame(np.random.randn(10, 3)),
+                                        pd.Series(np.random.randn(10)))
+    with pytest.raises(RuntimeError):
+        calculator.calculate_probabilistic_metrics(
+            proc_fxobs, LIST_OF_CATEGORIES, probabilistic._REQ_REF_FX,
+        )
+
+
+def test_calculate_probabilistic_metrics_no_reference_data(
+        single_prob_forecast_observation, create_processed_fxobs):
+    proc_fxobs = create_processed_fxobs(single_prob_forecast_observation,
+                                        pd.DataFrame(np.random.randn(10, 3)),
+                                        pd.Series(np.random.randn(10)))
+    proc_ref = create_processed_fxobs(single_prob_forecast_observation,
+                                      pd.DataFrame(),
+                                      pd.Series(np.random.randn(10)))
+    with pytest.raises(RuntimeError):
+        calculator.calculate_probabilistic_metrics(
+            proc_fxobs, LIST_OF_CATEGORIES, probabilistic._REQ_REF_FX,
+            proc_ref
+        )
+
+
+def test_calculate_probabilistic_metrics_bad_reference_interval_label(
+        single_prob_forecast_observation, create_processed_fxobs):
+    proc_fxobs = create_processed_fxobs(single_prob_forecast_observation,
+                                        pd.DataFrame(np.random.randn(10, 3)),
+                                        pd.Series(np.random.randn(10)))
+    proc_ref = create_processed_fxobs(single_prob_forecast_observation,
+                                      pd.DataFrame(np.random.randn(10, 3)),
+                                      pd.Series(np.random.randn(10)))
+    proc_ref = proc_ref.replace(interval_label='ending')
+    with pytest.raises(ValueError):
+        calculator.calculate_probabilistic_metrics(
+            proc_fxobs, LIST_OF_CATEGORIES, probabilistic._REQ_REF_FX,
+            proc_ref
+        )
+
+
+def test_calculate_probabilistic_metrics_interval_label_ending(
+        single_prob_forecast_observation, create_processed_fxobs,
+        create_dt_index):
+    proc_fxobs = create_processed_fxobs(
+        single_prob_forecast_observation,
+        pd.DataFrame(np.random.randn(10, 3), index=create_dt_index(10)),
+        pd.Series(np.random.randn(10), index=create_dt_index(10))
+    )
+    proc_fxobs = proc_fxobs.replace(interval_label='ending')
+    calculator.calculate_probabilistic_metrics(proc_fxobs, LIST_OF_CATEGORIES,
```

Comment:
there's no reason this test would fail even if interval label weren't handled properly, yes? is it easy enough to check one metric value that is different between ending and beginning?

Reply:
I changed it so the interval labels are explicitly different:

```python
    proc_fxobs = proc_fxobs.replace(interval_label='beginning')
    proc_ref = proc_ref.replace(interval_label='ending')
```

The test does fail for instance if both are `'beginning'`

```
        proc_fxobs = proc_fxobs.replace(interval_label='beginning')
        proc_ref = proc_ref.replace(interval_label='beginning')
        with pytest.raises(ValueError):
            calculator.calculate_probabilistic_metrics(
                proc_fxobs, LIST_OF_CATEGORIES, probabilistic._REQ_REF_FX,
>               proc_ref
            )
E           Failed: DID NOT RAISE <class 'ValueError'>
```